File: 22.10/219.go
func containsNearbyDuplicate(nums []int, k int) bool {
    a := make(map[int]int)
    for i, v := range(nums) {
        if a[v] != 0 && (i + 1 - a[v]) <= k {
            return true
        }
        a[v] = i + 1
    }
    return false
}

File: typescript/23.08/219.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
    const mem = {}
    for (let i=0; i<nums.length; i++) {
        if (mem[nums[i]] !== undefined && i - mem[nums[i]] <= k) {
            return true
        }
        mem[nums[i]] = i
    }
    return false
};
